app/code/Magento/Ui/Component/Form.php

```diff
@@ -63,8 +63,11 @@ public function getDataSourceData()
         $filter = $this->filterBuilder->setField($this->getContext()->getDataProvider()->getPrimaryFieldName())
             ->setValue($id)
             ->create();
-        $this->getContext()->getDataProvider()
-            ->addFilter($filter);
+        $dataProvider = $this->getContext()->getDataProvider();
+        if (isset($dataProvider->collection)) {
+            $this->getContext()->getDataProvider()
```

Comment:
I believe we don't need invoking  `$this->getContext()->getDataProvider()` once again since we have added `$dataProvider` variable